### ISSUE TITLE
Question about the training epoch of VQ-VAE

### ISSUE THREAD

**Issue by Heibaiii** [OpenMotionLab/MotionGPT#108]
Thank you for your great work!
I have a question about the different training epoch of VQ-VAE ( the motion tokenizer in paper )
In code, the epoch is set to be 999999.
![image](https://github.com/user-attachments/assets/e940ef26-d0cb-4fe9-9878-860b06e5947e)

In paper, the epoch is set to be 150000.
![image](https://github.com/user-attachments/assets/3f425d97-be70-4cc0-9284-c50e3c62ee9e)

In another question, the epoch is suggested to be 3000.
![image](https://github.com/user-attachments/assets/bfa4e932-479b-4098-a4c2-e3bf3c35738c)

What's more, I unzipped the pretrained model "motiongpt_s3_h3d.tar", and I found that the directory name is "epoch=119"
![image](https://github.com/user-attachments/assets/82b6fa5a-f091-43d1-bf66-125b38af3163)

And when I try to train a new VQ-VAE, I found that the reconstruction quality will be fine when the total loss is reduced to below 0.1
I want to know what's the difference between the different epoch of training VQ-VAE?
If 3000 epoch is good enough, why it should be trained to 150000 iterations?
I really appreciate your support and look forward to your response!

**Comment by DiamondGlassDrill** [comment]
@heibaiii what was your conclusion. Thanks in advance for sharing your thoughts and results.

**Comment by Heibaiii** [comment]
> [@Heibaiii](https://github.com/Heibaiii) what was your conclusion. Thanks in advance for sharing your thoughts and results.

I found that 'epoch' ≠ 'iteration'
For HumanML3D dataset, if batch_size = 256, 1 epoch ≈ 88 iteration because 23384 / 256 = 91.34 ≈ 88
23384 is the number of all motion data (include the mirror motion), when training, some wrong data are deleted, the rest data are used for training, so its 88 instead of 91.
I have tested epoch = 2000 and epoch = 3000, which indicate iteration =2000 *88 =176000 and iteration =3000*88=264000, the reconstruction quality are not too bad but not very good.
I'm still doing more experiments.
